src/library_addfunction.js

```diff
@@ -40,7 +45,7 @@ mergeInto(LibraryManager.library, {
 #if ASSERTIONS
       assert(sig[i] in typeNames, 'invalid signature char: ' + sig[i]);
 #endif
-      type.parameters.push(typeNames[sig[i]]);
+      Array.prototype.push.apply(type.parameters, typeNames[sig[i]]);
```

Comment:
Would it be more compact/efficient to avoid using arrays above and instead do 

#if !MEMORY64
// For i64 params we need two arguments.
if (sig[i] == 'j') type.parameters.push('i32');
#endif

I don't know if it is, WDYT?  I guess it depends on size and speed of  `Array.prototype.push.apply`

Reply:
I'm a little surprised this is needed.. since I would have through we would have run into this already.

Reply:
I was wondering about that too. Maybe our tests aren't exercising https://github.com/emscripten-core/emscripten/blob/40fb7d2071e439f1de614898b88518df582faa94/src/library_addfunction.js#L92-L94 ?